example/feature_demo/marshal_test.go

```diff
@@ -0,0 +1,64 @@
+package example
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/wrappers"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/infobloxopen/protoc-gen-gorm/types"
+)
+
+func TestUnmarshalTypes(t *testing.T) {
+	ts, _ := ptypes.TimestampProto(time.Date(2009, 11, 17, 20, 34, 58, 651387237, time.UTC))
+	marshaller := &runtime.JSONPb{OrigName: true, EmitDefaults: true}
+	for in, expected := range map[string]TestTypes{
+		`{}`: {},
+		`{"api_only_string":"important text"}`:            {ApiOnlyString: "important text"},
+		`{"numbers":[1,2,3,4]}`:                           {Numbers: []int32{1, 2, 3, 4}},
+		`{"optional_string":"something real"}`:            {OptionalString: &wrappers.StringValue{Value: "something real"}},
+		`{"becomes_int":"GOOD"}`:                          {BecomesInt: 1},
+		`{"uuid":"123abc-unchecked-uuid"}`:                {Uuid: &types.UUIDValue{Value: "123abc-unchecked-uuid"}},
+		`{"created_at":"2009-11-17T20:34:58.651387237Z"}`: {CreatedAt: ts},
+		`{"type_with_id_id":4}`:                           {TypeWithIdId: 4},
+		`{"json_field":{"top":[{"something":1},2]}}`:      {JsonField: &types.JSONValue{Value: `{"top":[{"something":1},2]}`}},
+	} {
+		tt := &TestTypes{}
+		err := marshaller.Unmarshal([]byte(in), tt)
+		if err != nil {
+			t.Error(err.Error())
```

Comment:
are there any error cases you want to test? If not, then you're basically never expecting to receive an error. In that case, you can use `t.Fatal(err.Error())` here instead